cmake/modules/hunter_windows_env_command.cmake

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2015 Sumedh Ghaisas, Ruslan Baratov
+# All rights reserved.
+
+include(hunter_windows_to_snf_filename)
+
+#Set command to change environment variables to 8.3 filenames to windows
+function(hunter_windows_env_command variable_name)
+  
+  hunter_windows_to_snf_filename(${CMAKE_C_COMPILER} "CMAKE_C_COMPILER_SNF")
+  hunter_windows_to_snf_filename(${CMAKE_CXX_COMPILER} "CMAKE_CXX_COMPILER_SNF")
+  hunter_windows_to_snf_filename(${CMAKE_RC_COMPILER} "CMAKE_RC_COMPILER_SNF")
+  
+  set(${variable_name} "set" "cc=${CMAKE_C_COMPILER_SNF}" "&&" "set" "cxx=${CMAKE_CXX_COMPILER_SNF}" "&&" "set" "rc=${CMAKE_RC_COMPILER_SNF}" PARENT_SCOPE)
```

Comment:
> Was there any reason behind keeping the qt cmake files in scripts?

Because it was only for Qt :) If other packages will use such approach too then it will be clear enough to figure out what layout we need and make a design decision.

> What do you think about making this a default for non cmake builds?

It will introduce some extra code that make harder to understand what's happenning so I guess it doesn't make sense to make this change for schemes where such trick not needed.

> I will shift the boost to this format as this is much cleaner approach
> Also my opinion is to shift the cmake files from scripts to individual folders

It's two separate features (with current one it's 3), let's do it one-by-one for easy testing purposes.